src/app/auth/services/auth.service.ts

```diff
@@ -1,30 +1,82 @@
-import { Injectable } from '@angular/core';
+import { Injectable } from "@angular/core";
+import { HttpClient } from "@angular/common/http";
+import { Observable, BehaviorSubject } from "rxjs";
+import { SessionStorageService } from "./session-storage.service";
+import { User } from "@app/shared/models/user.model";
+import { tap } from "rxjs";
+
+import { Router } from "@angular/router";
+import { UserStoreService } from "@app/user/services/user-store.service";
+
+const TOKEN = "SESSION_TOKEN";
 
 @Injectable({
-    providedIn: 'root'
+  providedIn: "root",
 })
 export class AuthService {
-    login(user: any) { // replace 'any' with the required interface
-        // Add your code here
-    }
+  private isAuthorized$$: BehaviorSubject<boolean>;
+  public isAuthorized$: Observable<boolean>;
+
+  constructor(
+    private http: HttpClient,
+    private sessionStorageService: SessionStorageService,
+    private router: Router,
+    private userStorageService: UserStoreService
+  ) {
+    this.isAuthorized$$ = new BehaviorSubject<boolean>(
+      this.sessionStorageService.hasToken()
+    );
+    this.isAuthorized$ = this.isAuthorized$$.asObservable();
+  }
+  login(user: User): Observable<any> {
+    // replace 'any' with the required interface
+    // Add your code here
+
+    return this.http.post<any>(this.getLoginUrl(), user).pipe(
+      tap((response) => {
+        if (response && response.result) {
+          this.sessionStorageService.setToken(response.result);
+          this.isAuthorized$$.next(true);
+
+          this.router.navigate(["/courses"]);
+        }
+      })
+    );
+  }
+
+  logout(): void {
+    // Add your code here
+    this.sessionStorageService.deleteToken();
+    this.isAuthorized$$.next(false);
+    this.router.navigate(["/login"]);
+  }
 
-    logout() {
-        // Add your code here
-    }
+  register(user: User): Observable<any> {
+    // replace 'any' with the required interface
+    // Add your code here
+    return this.http.post<any>(this.getRegisterUrl(), user);
+  }
 
-    register(user: any) { // replace 'any' with the required interface
-        // Add your code here
-    }
+  get isAuthorised() {
+    // Add your code here. Get isAuthorized$$ value
+    return this.isAuthorized$$.getValue();
```

Comment:
Probably you wan to return the public member of this class instead of reading the value of the private member.

Reply:
The task write and said Get isAuthorized$$ value, if I return isAuthorized$ member with the getter what will be the difference between this getter method and the public isAuthorised$ member?